consensus/core/src/leader_schedule.rs

```diff
@@ -269,20 +266,51 @@ impl LeaderSwapTable {
     // The `swap_stake_threshold` should be in the range of [0 - 33].
     pub(crate) fn new(
         context: Arc<Context>,
+        commit_index: CommitIndex,
         reputation_scores: ReputationScores,
+    ) -> Self {
+        let swap_stake_threshold = context
+            .protocol_config
+            .consensus_bad_nodes_stake_threshold();
+        Self::new_inner(
+            context,
+            swap_stake_threshold,
+            commit_index,
+            reputation_scores,
+        )
+    }
+
+    fn new_inner(
```

Comment:
I understand that this is for testing convenience right? Otherwise you can configure the tests with stake threshold like 
```
let mut context = Context::new_for_test(4).0;
        context
            .protocol_config
            .set_consensus_bad_nodes_stake_threshold(33);
        let context = Arc::new(context);
```